mysql/datadog_checks/mysql/mysql.py

```diff
@@ -103,6 +104,9 @@ def check(self, _):
                 self.version = get_version(db)
                 self._send_metadata()
 
+                # Supplement tags with runtime data
+                self.config.tags += self._get_runtime_tags(db)
```

Comment:
That's a memory leak right there, you are appending new tags on every check run. As a rule of thumb, don't modify self.config in the check method